public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useMemo } from 'react';
+import {
+  DataTransformerID,
+  standardTransformers,
+  TransformerRegistyItem,
+  TransformerUIProps,
+  getFieldDisplayName,
+  Field,
+  DataFrame,
+} from '@grafana/data';
+import { getAllFieldNamesFromDataFrames } from './OrganizeFieldsTransformerEditor';
+import { Select, Button, Input } from '@grafana/ui';
+
+import {
+  FilterByValueTransformerOptions,
+  ValueFilter,
+} from '@grafana/data/src/transformations/transformers/filterByValue';
+
+import { valueFiltersRegistry, ValueFilterID } from '@grafana/data/src/transformations/valueFilters';
+
+function FilterSelectorRow(props: any) {
+  const { fieldNameOptions, onDelete, onConfigChange, config, fieldType } = props;
+
+  console.log('props', props);
+
+  // Find filter types that fit the chosen field type
+  let filterTypeOptions = valueFiltersRegistry
+    .list()
+    .filter(element => {
+      if (element.supportedFieldTypes) {
+        return !element.supportedFieldTypes ? true : element.supportedFieldTypes.includes(fieldType);
+      } else {
+        return true;
+      }
+    })
+    .map(item => ({ value: item.id, label: item.name, description: item.description }));
+
+  let filterInfo = valueFiltersRegistry.get(config.filterType);
+  let filterValid = filterInfo.getInstance({
+    filterExpression: config.filterExpression,
+    fieldType: fieldType,
+  }).isValid;
+
+  let fieldNameInvalid =
+    config.fieldName === null ||
+    fieldNameOptions.filter((item: Record<string, any>) => item.value === config.fieldName).length === 0;
+  let filterTypeInvalid =
+    !fieldNameInvalid && filterInfo.supportedFieldTypes && !filterInfo.supportedFieldTypes.includes(fieldType);
+  let filterExpressionInvalid = !fieldNameInvalid && !filterTypeInvalid && !filterValid;
+
+  let filterOptionsInput = null;
+  if (filterInfo.placeholder) {
+    filterOptionsInput = (
+      <>
+        <Input
+          className="flex-grow-1"
+          invalid={filterExpressionInvalid}
+          defaultValue={config.filterExpression}
+          placeholder={filterInfo.placeholder}
+          onBlur={event => {
+            onConfigChange({ ...config, filterExpression: event.currentTarget.value });
+          }}
+        />
+      </>
+    );
+  }
```

Comment:
You could do the check where you want to render it instead of doing it this way. The benefit of doing the check where you want to render it is that you don't have to jump up and down when you read the code.